openqa_bugfetcher/issues/bugzilla_issue.py

```diff
@@ -1,38 +1,44 @@
 import html
-import json
-from collections import OrderedDict
 
 import requests
 
 from openqa_bugfetcher.issues import BaseIssue
 
 
+# error codes:
+# https://github.com/bugzilla/bugzilla/blob/master/Bugzilla/WebService/Constants.pm#L160
+
+
 class BugzillaIssue(BaseIssue):
     prefixes = {"boo", "bnc", "bsc"}
 
     def fetch(self, conf):
         issue_id = self.bugid.split("#")[1]
         if "user" in conf["bugzilla"]:
 
-            def json_rpc_get(url, method, params):
-                get_params = OrderedDict({"method": method, "params": json.dumps([params])})
-                cred = conf["bugzilla"]
-                return requests.get(url, params=get_params, auth=(cred["user"], cred["pass"]))
+            def rest_get_bug(issue_id):
+                url = f"https://bugzilla.suse.com/rest/bug/{issue_id}"
+                get_params = {}
+                cfg = conf["bugzilla"]
+                if cfg.get("api_key"):
+                    get_params["api_key"] = cfg["api_key"]
+                return requests.get(url, params=get_params, timeout=10)
 
-            url = "https://apibugzilla.suse.com/jsonrpc.cgi"
-
-            req = json_rpc_get(url, "Bug.get", {"ids": [issue_id]})
+            req = rest_get_bug(issue_id)
             assert req.status_code != 401, "Wrong auth for Bugzilla"
             assert req.status_code != 403, "Insufficient permission to access this bug (or the API at all)"
-            assert req.ok
+            assert req.ok, f"Server Status != 200: {req.text}"
             data = req.json()
             assert data, "Empty JSON Object"
-            if data["error"] and data["error"]["code"] in (100, 101):
-                self.existing = False
-            elif data["error"] and data["error"]["code"] == 102:
-                raise AssertionError("Insufficient permission to access this bug")
+            if data.get("error"):
+                if data["code"] in (100, 101):
```

Comment:
Just checking - did it really change from `data["error"]["code"]` to `data["code"]`?

Reply:
yes.

```
curl -s https://bugzilla-next.suse.de/rest/bug/1204384 | json_pp
{
   "code" : 102,
   "documentation" : "https://bugzilla.readthedocs.org/en/5.0/api/",
   "error" : true,
   "message" : "You are not authorized to access bug #1204384. To see this bug, you must first log in to an account with the appropriate permissions."
}
```